packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -212,41 +219,13 @@ class _ProductQueryPageState extends State<ProductQueryPage>
                 return _getTopMessagesCard();
               }
               index--;
+              // TODO(monsieurtanuki): maybe call it earlier, like for first unknown page index - 5?
               if (index >= _model.displayBarcodes.length) {
-                // final button
-                final int already = _model.displayBarcodes.length;
-                final int totalSize =
-                    _model.supplier.partialProductList.totalSize;
-                final int next = max(
-                  0,
-                  min(
-                    _model.supplier.productQuery.pageSize,
-                    totalSize - already,
-                  ),
-                );
-                final Widget child;
-                if (next == 0) {
-                  child = EMPTY_WIDGET;
-                } else {
-                  child = SmoothLargeButtonWithIcon(
-                    text: appLocalizations.product_search_button_download_more(
-                      next,
-                      already,
-                      totalSize,
-                    ),
-                    icon: Icons.download_rounded,
-                    onPressed: () async =>
-                        _loadAndRefreshDisplay(_model.loadNextPage()),
-                  );
-                }
-                return Padding(
-                  padding: const EdgeInsetsDirectional.only(
-                    bottom: 90.0,
-                    start: VERY_LARGE_SPACE,
-                    end: VERY_LARGE_SPACE,
-                  ),
-                  child: child,
-                );
+                _downloadNextPage();
+              }
+              if (index >= _model.displayBarcodes.length) {
+                // TODO(monsieurtanuki): maybe display something specific for data being downloaded (the next page) and unknown data (beyond next page)
+                return const SmoothProductCardTemplate();
               }
```

Comment:
One thing, the template has no common loading animation it is just a plan thing, if it takes over a certain amount of time and the user can scroll as long as he want's it could create the illusion that something is broken, therefor I'd suggest to at max show only one template at a time.

Reply:
it's usually solved using a shimmering effect

Reply:
Yeah but I'd still prefer a single one and as we know monsieurtanuki isn't a friend of fancy animations so thats something we can always add later

Reply:
ok